### ISSUE TITLE
[Bug]: git obsidian error: Cannot run git command

### ISSUE THREAD

**Issue by Chen-913** [denolehov/obsidian-git#235]
### Describe the bug

I installed Obsidian git in Obsidian's community plug-in. After installation, I set the remote warehouse address, which can be used normally and successfully pushed to the remote warehouse. However, when I turn on the next time and open obsidian, it prompts me that I can't work. I try to find the answer in the document, but it can't be solved

My system is windows 10

### Relevant errors (if available)

```shell
Error: spawnSync https://gitee.com/xxxx/xx ENOENT
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at __node_internal_errnoException (node:internal/errors:594:12)
    at Object.spawnSync (node:internal/child_process:1083:20)
    at spawnSync (node:child_process:821:24)
    at SimpleGit.isGitInstalled (eval at <anonymous> (app://obsidian.md/app.js:1:1), <anonymous>:11171:57)
    at SimpleGit.eval (eval at <anonymous> (app://obsidian.md/app.js:1:1), <anonymous>:10830:16)
    at Generator.next (<anonymous>)
    at eval (eval at <anonymous> (app://obsidian.md/app.js:1:1), <anonymous>:66:61)
    at new Promise (<anonymous>)
    at __async (eval at <anonymous> (app://obsidian.md/app.js:1:1), <anonymous>:50:10)
    at SimpleGit.setGitInstance (eval at <anonymous> (app://obsidian.md/app.js:1:1), <anonymous>:10829:12)
    at ObsidianGit.eval (eval at <anonymous> (app://obsidian.md/app.js:1:1), <anonymous>:14415:33)
    at Generator.next (<anonymous>)
    at eval (eval at <anonymous> (app://obsidian.md/app.js:1:1), <anonymous>:66:61)
    at new Promise (<anonymous>)
    at __async (eval at <anonymous> (app://obsidian.md/app.js:1:1), <anonymous>:50:10)
    at ObsidianGit.init (eval at <anonymous> (app://obsidian.md/app.js:1:1), <anonymous>:14411:12)
    at eval (eval at <anonymous> (app://obsidian.md/app.js:1:1), <anonymous>:14358:51)
    at t.<anonymous> (app://obsidian.md/app.js:1:1443816)
    at app://obsidian.md/app.js:1:234991
    at Object.next (app://obsidian.md/app.js:1:235096)
    at s (app://obsidian.md/app.js:1:233835)


------------------------------------------


Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'status')
    at SimpleGit.eval (eval at <anonymous> (app.js:1:1), <anonymous>:11061:37)
    at Generator.next (<anonymous>)
    at eval (eval at <anonymous> (app.js:1:1), <anonymous>:66:61)
    at new Promise (<anonymous>)
    at __async (eval at <anonymous> (app.js:1:1), <anonymous>:50:10)
    at SimpleGit.branchInfo (eval at <anonymous> (app.js:1:1), <anonymous>:11060:12)
    at ObsidianGitSettingsTab.eval (eval at <anonymous> (app.js:1:1), <anonymous>:6883:50)
    at Generator.next (<anonymous>)
    at eval (eval at <anonymous> (app.js:1:1), <anonymous>:66:61)
    at new Promise (<anonymous>)
```

### Steps to reproduce

1. Install plug-ins in the plug-in community
2. Reset computer 
3. Open Obsidian

### Operating system

Windows

### Plugin version

1.25.2

**Comment by Vinzent03** [comment]
Did you enter `https://gitee.com/xxxx/xx` as `Custom Git binary path`? That's not supposed to be an url. It's for setting the path to the git binary on your machine, if it's not in your `PATH`.
I think you are searching for setting the remote url. Please use the command `Edit Remotes`.

**Comment by BorisZubchenko** [comment]
Having the same issue. Crashes on [isGitInstalled](https://github.com/denolehov/obsidian-git/blame/e3a24650b949ec78a1d87bd02f5668bd3b039489/src/simpleGit.ts#L391):
- created a GitHub private repo
- in the vault, `git init`
- `git add .`
- `git commit -m "Initial commit"`
- `git remote add origin https://github.com/BorisZubchenko/vault.git`
- `git push -u origin master`
- put my credentials
- restarted Obsidian

```
VM169:11181 Error: spawnSync git ENOENT
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at __node_internal_errnoException (node:internal/errors:594:12)
    at Object.spawnSync (node:internal/child_process:1083:20)
    at spawnSync (node:child_process:821:24)
    at SimpleGit.isGitInstalled (eval at <anonymous> (app.js:1:1494667), <anonymous>:11177:57)
    at SimpleGit.eval (eval at <anonymous> (app.js:1:1494667), <anonymous>:10830:16)
    at Generator.next (<anonymous>)
    at eval (eval at <anonymous> (app.js:1:1494667), <anonymous>:66:61)
    at new Promise (<anonymous>)
    at __async (eval at <anonymous> (app.js:1:1494667), <anonymous>:50:10)
    at SimpleGit.setGitInstance (eval at <anonymous> (app.js:1:1494667), <anonymous>:10829:12)
    at ObsidianGit.eval (eval at <anonymous> (app.js:1:1494667), <anonymous>:14421:33)
    at Generator.next (<anonymous>)
    at eval (eval at <anonymous> (app.js:1:1494667), <anonymous>:66:61)
    at new Promise (<anonymous>)
    at __async (eval at <anonymous> (app.js:1:1494667), <anonymous>:50:10)
    at ObsidianGit.init (eval at <anonymous> (app.js:1:1494667), <anonymous>:14417:12)
    at eval (eval at <anonymous> (app.js:1:1494667), <anonymous>:14364:51)
    at t.<anonymous> (app.js:1:1443816)
    at app.js:1:234991
    at Object.next (app.js:1:235096)
    at s (app.js:1:233835)
```
```
Error: spawnSync git ENOENT
    at __node_internal_captureLargerStackTrace (node:internal/errors:464:5)
    at __node_internal_errnoException (node:internal/errors:594:12)
    at Object.spawnSync (node:internal/child_process:1083:20)
    at spawnSync (node:child_process:821:24)
    at SimpleGit.isGitInstalled (eval at <anonymous> (app.js:1:1494667), <anonymous>:11177:57)
    at SimpleGit.eval (eval at <anonymous> (app.js:1:1494667), <anonymous>:11056:17)
    at Generator.next (<anonymous>)
    at eval (eval at <anonymous> (app.js:1:1494667), <anonymous>:66:61)
    at new Promise (<anonymous>)
    at __async (eval at <anonymous> (app.js:1:1494667), <anonymous>:50:10)
    at SimpleGit.checkRequirements (eval at <anonymous> (app.js:1:1494667), <anonymous>:11055:12)
    at ObsidianGit.eval (eval at <anonymous> (app.js:1:1494667), <anonymous>:14423:46)
    at Generator.next (<anonymous>)
    at fulfilled (eval at <anonymous> (app.js:1:1494667), <anonymous>:53:24)
```

Also, select branch dropdown is empty.

Ubuntu 22.04
Plugin version 1.25.3

**Comment by Vinzent03** [comment]
@BorisZubchenko 
How did you install Obsidian? https://github.com/denolehov/obsidian-git/wiki/Installation#linux

**Comment by BorisZubchenko** [comment]
@Vinzent03, via Snap. Now I see why it doesn't work. Thank you and sorry for being inconsiderate.

**Comment by Vinzent03** [comment]
@BorisZubchenko No Problem. It's somewhat hidden currently. I should state it more clearly in the README.

**Comment by Vinzent03** [comment]
@Chen-913 
Closing because of no response. Just comment if you have still problems, and I'm open to reopen the issue.

**Comment by Chen-913** [comment]
> Did you enter `https://gitee.com/xxxx/xx` as `Custom Git binary path`? That's not supposed to be an url. It's for setting the path to the git binary on your machine, if it's not in your `PATH`. I think you are searching for setting the remote url. Please use the command `Edit Remotes`.

I think I should set git binary path correctly, because I can successfully push to my remote warehouse when I set it for the first time.but after that I restart my computer. It doesn't work normally until now. As you said, I open the control panel with `Ctrl + P` and enter edit remote. After I enter edit remote, I still prompt `Can not run git commond` in the upper right corner, just like I just opened obsidian
in addition `https://gitee.com/xx/xx` I did desensitization because it was not an open warehouse
